code/modules/surgery/organs/augments_internal.dm

```diff
@@ -457,6 +454,87 @@
 
+/obj/item/organ/internal/cyberimp/chest/ipc_repair
+	name = "Reactive Repair Implant"
+	desc = "This implant reworks the IPC frame, in order to incorporate materials that return to their original shape after being damaged. Requires power to function."
+	implant_color = "#0ac0d8"
+	origin_tech = "materials=4;programming=4;biotech=4;magnets=4;engineering=4"
+	slot = "stomach" //Can't have a nutriment pump with it.
+	requires_machine_person = TRUE
+
+/obj/item/organ/internal/cyberimp/chest/ipc_repair/on_life()
+	if(crit_fail)
+		return
+	if(owner.maxHealth == owner.health)
+		owner.adjust_nutrition(-0.5)
+		return //Passive damage scanning
+
+	owner.adjustBruteLoss(-0.5, robotic = TRUE)
+	owner.adjustFireLoss(-0.5, robotic = TRUE)
+	owner.adjust_nutrition(-4) //Very power inefficent. Hope you got an APC nearby.
+
+/obj/item/organ/internal/cyberimp/chest/ipc_repair/emp_act(severity)
+	if(!owner || emp_proof || crit_fail)
+		return
+	crit_fail = TRUE
+	addtimer(VARSET_CALLBACK(src, crit_fail, FALSE), 30 SECONDS / severity)
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints
+	name = "IPC ER-OR Joint Implant"
+	desc = "This is a basetype. Notify a coder!"
+	implant_color = "#eeff00"
+	origin_tech = "materials=5;programming=4;biotech=4"
+	slot = "joints"
+	requires_machine_person = TRUE
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/magnetic_joints
+	name = "Magnetic Joints Implant"
+	desc = "This implant modifies IPC joints to use magnets, allowing easy re-attachment and fluid movement."
+	implant_color = "#670db1"
+	origin_tech = "materials=4;programming=4;biotech=4;magnets=4;engineering=4"
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/magnetic_joints/emp_act(severity)
+	if(!owner || emp_proof)
+		return
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		to_chat(H, "<span class='userdanger'>Your magnetic joints lose power!</span>")
+		for(var/obj/item/organ/external/E in H.bodyparts)
+			if(E.body_part != UPPER_TORSO && E.body_part != LOWER_TORSO)
+				E.droplimb(TRUE) //lego disasemble sound
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/magnetic_joints/insert(mob/living/carbon/M, special = FALSE)
+	ADD_TRAIT(M, TRAIT_IPC_JOINTS_MAG, "ipc_joint[UID()]")
+	return ..()
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/magnetic_joints/remove(mob/living/carbon/M, special = FALSE)
+	REMOVE_TRAIT(M, TRAIT_IPC_JOINTS_MAG, "ipc_joint[UID()]")
+	return ..()
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/sealed
+	name = "Sealed Joints Implant"
+	desc = "This implant seals and reinforces IPC joints, securing the limbs better for industrial work, though prone to locking up."
+	implant_color = "#b10d0d"
+	origin_tech = "materials=4;programming=4;biotech=4;engineering=4;combat=4;"
+
+/obj/item/organ/internal/cyberimp/chest/ipc_joints/sealed/emp_act(severity)
+	if(!owner || emp_proof)
+		return
+	var/weaken_time = (10 + (severity - 1 ? 0 : 10)) STATUS_EFFECT_CONSTANT
```

Comment:
This was a holdover from the status effect refactor for things that would have been annoying to recalculate into seconds.

```suggestion
	var/weaken_time = (10 + (severity - 1 ? 0 : 10)) SECONDS
```